src/wallet/gtest/test_wallet.cpp

```diff
@@ -138,17 +138,25 @@ TEST(WalletTests, WalletNetworkSerialization) {
     // create a new testnet wallet and generate a seed
     CWallet wallet(Params(), "wallet.dat");
     wallet.InitLoadWallet(Params(), true);
-    wallet.GenerateNewSeed();
```

Comment:
The call to `initLoadWallet()` on the previous line called `GenerateNewSeed()`, so this was adding a second seed, and _that_ is why the wallet failed to load below -- not because of the network mismatch.